hi-ml-histopathology/src/histopathology/configs/classification/BaseMIL.py

```diff
@@ -92,6 +96,15 @@ def setup(self) -> None:
         self.encoder = self.get_encoder()
         if not self.is_finetune:
             self.encoder.eval()
+        # Fine-tuning requires tiles to be loaded on-the-fly, hence, caching is disabled by default.
+        if self.is_finetune:
+            self.is_caching = False
+        if self.is_caching:
+            self.cache_mode = CacheMode.MEMORY
+            self.precache_location = CacheLocation.CPU
+        else:
```

Comment:
How about having `is_caching` override the caching options only in the negative case? That way, users can still configure `cache_mode` and `precache_location` if `is_caching is False`.

Reply:
I agree, removed the positive case now. I have changed the default for `precache_location` in BaseMIL